cli_tools/gce_vm_image_import/importer/inflater.go

```diff
@@ -41,6 +42,103 @@ const (
 	defaultInflationDiskSizeGB = 10
 )
 
+// inflaterFacade implements an inflater using other concrete implementations.
+type inflaterFacade struct {
+	mainInflater   inflater
+	shadowInflater inflater
+}
+
+// signals to control the verification towards shadow inflater
+const (
+	sigMainInflaterDone   = "main done"
+	sigMainInflaterErr    = "main err"
+	sigShadowInflaterDone = "shadow done"
+	sigShadowInflaterErr  = "shadow err"
+)
+
+func (facade *inflaterFacade) inflate() (persistentDisk, error) {
+	inflaterChan := make(chan string)
+
+	// Launch main inflater.
+	var pd persistentDisk
+	var err error
+	go func() {
+		pd, err = facade.mainInflater.inflate()
+		if err != nil {
+			inflaterChan <- sigMainInflaterErr
+		} else {
+			inflaterChan <- sigMainInflaterDone
+		}
+	}()
+
+	// Launch shadow inflater.
+	var shadowPd persistentDisk
+	var shadowErr error
+	go func() {
+		shadowPd, shadowErr = facade.shadowInflater.inflate()
+		if shadowErr != nil {
+			inflaterChan <- sigShadowInflaterErr
+		} else {
+			inflaterChan <- sigShadowInflaterDone
+		}
+	}()
+
+	// Return early if main inflater finished first.
+	result := <-inflaterChan
```

Comment:
should this be called shadowResult?

Reply:
this one can be either main result or shadow result. That is why I use a if-condition to distinguish them